src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs

```diff
@@ -222,6 +232,39 @@ private static GitPullRequest CreatePullRequest(string sourceBranch, string targ
             throw new InvalidOperationException("Could not find drop path");
         }
 
+        private static async Task<string> DownloadArtifactsAsync(BuildHttpClient buildClient, Build build, BuildArtifact artifact, CancellationToken cancellationToken)
+        {
+            var tempDirectory = Path.Combine(Path.GetTempPath(), string.Concat(Options.InsertionName, Options.BranchName).Replace(" ", "_").Replace("/","_"));
+            if (Directory.Exists(tempDirectory))
+            {
+                // Be judicious and clean up old artifacts so we do not eat up memory on the scheduler machine.
+                // Sometime FileSystem APIs do not create a directory immediately after a deletion - Looks like there is a race.
```

Comment:
Suspicious comment is suspicious. I wouldn't be so quick to blame file system APIs...

Reply:
[This](https://stackoverflow.com/questions/32593552/using-directory-delete-and-directory-createdirectory-to-overwrite-a-folder) is the only reference I found that sympathizes with this change. Ran into this twice when testing locally.